### PR TITLE
fix default solver interpolation

### PR DESCRIPTION
**Pull request by oscardssmith**
Fixes issue observed by @TorkelE where the default solver wasn't interpolating properly.

### CODE REVIEW

File: src/dense/generic_dense.jl [hunk 1]
```diff
@@ -459,25 +649,32 @@ function ode_interpolation!(vals, tvals, id::I, idxs, deriv::D, p,
...
+                    cache_i₊ = cache.cache2
+                elseif current_alg == 3
+                    cache_i₊ = cache.cache3
+                elseif cerrent_alg == 4
+                    cache_i₊ = cache.cache4
+                elseif current_alg == 5
+                    cache_i₊ = cache.cache5
+                elseif current_alg == 6
+                    cache_i₊ = cache.cache6
+                end
```
Comment (ChrisRackauckas) [inline src/dense/generic_dense.jl:667]: Is this actually used anywhere?

File: src/dense/generic_dense.jl [hunk 2]
```diff
@@ -415,9 +590,24 @@ function ode_interpolation!(vals, tvals, id::I, idxs, deriv::D, p,
...
         cache_i₊ = cache.caches[current_alg]
+    elseif cache isa DefaultCache
+        current_alg = id.alg_choice[i₊]
+        if current_alg == 1
+            cache_i₊ = cache.cache1
+        elseif current_alg == 2
+            cache_i₊ = cache.cache2
+        elseif current_alg == 3
+            cache_i₊ = cache.cache3
+        elseif cerrent_alg == 4
```
Comment (ranocha) [inline src/dense/generic_dense.jl:604]:
```suggestion
        elseif current_alg == 4
```

File: src/dense/generic_dense.jl [hunk 3]
```diff
@@ -459,25 +649,32 @@ function ode_interpolation!(vals, tvals, id::I, idxs, deriv::D, p,
...
+        elseif cache isa DefaultCache
+            if current_alg != id.alg_choice[i₊] # switched algorithm
+                current_alg = id.alg_choice[i₊]
+                if current_alg == 1
+                    cache_i₊ = cache.cache1
+                elseif current_alg == 2
+                    cache_i₊ = cache.cache2
+                elseif current_alg == 3
+                    cache_i₊ = cache.cache3
+                elseif cerrent_alg == 4
```
Comment (ranocha) [inline src/dense/generic_dense.jl:661]:
```suggestion
                elseif current_alg == 4
```

File: src/dense/generic_dense.jl (hunk 1 above)
Comment (oscardssmith) [reply src/dense/generic_dense.jl]: yeah. It's used further down in the function. Note that I removed the `else` so that it now runs for all cache types.

Comment (ChrisRackauckas) [reply src/dense/generic_dense.jl]: Don't you need to use it in the same part of the for loop to avoid the dynamism?

Comment (oscardssmith) [reply src/dense/generic_dense.jl]: I guess that would be better. I was following the design for Composite, but I guess that doesn't make as much sense for default.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Merge when tests pass.

**Review by tomaklutfu — COMMENTED** [review]
`C` is unused here. Did you intend to do `cache::Tuple{C}`?

**Comment by ChrisRackauckas** [comment]
Oh this right here? 

https://github.com/SciML/OrdinaryDiffEq.jl/pull/2245/files/c5e292415127e344d7084acd1cee86001e05bd36#diff-880197818e0d2bb2dab8f58cc3be7df428ac18e981932f5de86923a298405d66R496

Yeah that looks like a typo

**Comment by oscardssmith** [comment]
good catch. Not a problem, but unintended and easy to fix.https://github.com/SciML/OrdinaryDiffEq.jl/pull/2248

**Pull request merged**